railties/lib/rails/commands/server/server_command.rb

```diff
@@ -56,10 +57,19 @@ def default_options
     end
 
     def served_url
-      "#{options[:SSLEnable] ? 'https' : 'http'}://#{options[:Host]}:#{options[:Port]}" unless use_puma?
+      server_url unless use_puma?
     end
 
     private
+      def server_url
+        "#{options[:SSLEnable] ? 'https' : 'http'}://#{options[:Host]}:#{options[:Port]}"
```

Comment:
Do we actually care about SSL? Sounds to be we could always use HTTP given it's expected to be on localhost.

Or am I missing something?

Reply:
It would be unusual, but there may be situations where people want to use SSL locally, e.g. https://madeintandem.com/blog/rails-local-development-https-using-self-signed-ssl-certificate/

Reply:
No, I get that people may want to have SSL locally to better test / develop their app. What I question is the need of using an SSL socket for the language server protocol.

But as mentioned later, I also question whether relying on Rails server is the best Idea.